packages/loader/src/loader.ts

```diff
@@ -165,9 +179,23 @@ export class DocumentLoader extends DataLoader<DocID, ModelInstanceDocument> {
    * Load a ModelInstanceDocument from the cache (if enabled) or remotely.
    */
   async load<T extends Record<string, any> = Record<string, any>>(
-    id: DocID,
+    key: LoadKey,
+  ): Promise<ModelInstanceDocument<T>> {
+    return (await super.load(key)) as ModelInstanceDocument<T>
+  }
+
+  /**
+   * Load a deterministic stream and add it to the cache.
+   */
+  async _loadDeterministic<T extends Record<string, any> = Record<string, any>>(
+    meta: GenesisMetadata,
+    options: CreateOpts = {},
   ): Promise<ModelInstanceDocument<T>> {
-    return (await super.load(id)) as ModelInstanceDocument<T>
+    const opts = { ...DEFAULT_DETERMINISTIC_OPTIONS, ...options }
+    const key = await this._getDeterministicKey(meta)
+    const stream = await this.load<T>({ ...key, opts })
```

Comment:
I don't really feel like I understand how DataLoader works well enough to know if this is safe or not.  It feels fishy to me, and I know the latency test you shared with me the other day was showing this code path to still have high latency (ie not be respecting the `opts` properly).  Are you sure this is going to pass the `CreateOpts` properly?